src/note_item.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrash, faExpandArrowsAlt, faPencilAlt } from '@fortawesome/free-solid-svg-icons';
+import Draggable from 'react-draggable';
+import marked from 'marked';
+
+
+class NoteItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isEditing: false, text: this.props.note.text, title: this.props.note.title, x: this.props.note.x, y: this.props.note.y,
+    };
+  }
+
+  editBoxSubmit = (event) => {
+    const updates = {};
+    updates[this.props.id] = {
+      title: this.state.title, text: this.state.text, x: this.state.x, y: this.state.y,
+    };
+    this.props.update(updates);
+    this.setState({ isEditing: false });
+  }
+
+  myhandleDrag = (e, data) => {
+    this.setState({
+      x: data.x,
+      y: data.y,
+    });
+    const updates = {};
+    updates[this.props.id] = {
+      title: this.state.title, text: this.state.text, x: this.state.x, y: this.state.y,
```

Comment:
For example, here you can just directly send x and y up instead of using local state